core/src/main/java/org/jruby/runtime/load/LibrarySearcher.java

```diff
@@ -133,6 +132,20 @@ private FoundLibrary findResourceLibrary(String baseName, String suffix) {
           return findFileResource(baseName, suffix);
         }
 
+        // ruby does not load a relative path unless the current working directory is in $LOAD_PATH
+        if (baseName.startsWith("./") || loadService.loadPath.contains(".")) {
```

Comment:
./ just bypasses the LOAD_PATH - yes
but if I do not find it with "." on the findFileResourceWithLoadPath then the classloader will find it. since the application classloader has the PWD (the one from the java world) on its classpath.

so the test of "." is needed to not fall into the classloader search and find it there. and the classloader finds always come with a "uri:classloader:" prefix which is not wrong but strange for files from the filesystem.